### ISSUE TITLE
Random failure: "web3.exceptions.TimeExhausted" in test_polygon::test_ocean_tx__create_url_asset

### ISSUE THREAD

**Issue by trentmc** [oceanprotocol/ocean.py#1017]
NOTE: the failing test is [turned off](https://github.com/oceanprotocol/ocean.py/commit/faa9ab720e99f169ae150392e850abbbe28105e1), until we fix this issue. 

----

**Describe the bug**

This occurs randomly on some CI runs. E.g. this one: https://github.com/oceanprotocol/ocean.py/actions/runs/3255762872/jobs/5345414345

**Proposed solution**

Do like test_mumbai.py's try/except where `# it can get away with "<error x>" errors, but not others`. In our case, error x = `"TimeExhausted"`.

**Details: Full log**

Here's a full log:
```text
Run mkcodes --github --output tests/generated-readmes/test_{name}.{ext} READMEs
============================= test session starts ==============================
platform linux -- Python 3.8.14, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/runner/work/ocean.py/ocean.py, configfile: pytest.ini
plugins: web3-5.28.0, env-0.6.2
collected 249 items / 8 deselected / 241 selected

ocean_lib/aquarius/test/test_aquarius.py ....                            [  1%]
..
tests/integration/remote/test_polygon.py F                               [ 91%]
tests/readmes/test_readmes.py .....................                      [100%]

=================================== FAILURES ===================================
_______________________ test_ocean_tx__create_url_asset ________________________

self = <web3.eth.Eth object at 0x7f3c16fc50a0>
transaction_hash = HexBytes('0xf7e9c54c1c6e3b6830daddf8351a281989744d5c50e25e91ccec957d09ce7fc2')
timeout = 60, poll_latency = 0.1

    def wait_for_transaction_receipt(
        self, transaction_hash: _Hash32, timeout: float = 120, poll_latency: float = 0.1
    ) -> TxReceipt:
        try:
            with Timeout(timeout) as _timeout:
                while True:
                    try:
                        tx_receipt = self._get_transaction_receipt(transaction_hash)
                    except TransactionNotFound:
                        tx_receipt = None
                    if tx_receipt is not None:
                        break
>                   _timeout.sleep(poll_latency)

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/web3/eth.py:770: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Timeout(60), seconds = 0.1

    def sleep(self, seconds: float) -> None:
        time.sleep(seconds)
>       self.check()

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/web3/_utils/threads.py:89: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Timeout(60)

    def check(self) -> None:
        if self.is_running is None:
            raise ValueError("Timeout has not been started")
        elif self.is_running is False:
            raise ValueError("Timeout has already been cancelled")
        elif self.seconds is None:
            return
        elif time.time() > self.expire_at:
            self.is_running = False
            if isinstance(self.exception, type):
                raise self.exception(str(self))
            elif isinstance(self.exception, Exception):
                raise self.exception
            else:
>               raise self
E               web3._utils.threads.Timeout: 60 seconds

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/web3/_utils/threads.py:82: Timeout

During handling of the above exception, another exception occurred:

tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_ocean_tx__create_url_asse0')

    @pytest.mark.integration
    def test_ocean_tx__create_url_asset(tmp_path):
        """On Polygon, do the Ocean txs for create_url_asset(). Captures issue:https://github.com/oceanprotocol/ocean.py/issues/1007#issuecomment-1276286245"""
    
        # setup
        config = _remote_config_polygon(tmp_path)
        ocean = Ocean(config)
        (alice_wallet, _) = get_wallets(ocean)
    
        # Alice call create_url_asset
        # avoid "replacement transaction underpriced" error: make each tx diff't
        name = random_chars()
        url = "https://arweave.net/qctEbPb3CjvU8LmV3G_mynX74eCxo1domFQIlOBH1xU"
        try:  # it can get away with "insufficient funds" errors, but not others
            print("Call create_url_asset(), and wait for it to complete...")
>           (_, datatoken, _) = ocean.assets.create_url_asset(
                name, url, alice_wallet, wait_for_aqua=False
            )

tests/integration/remote/test_polygon.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/ocean/ocean_assets.py:270: in create_url_asset
    return self._create1(name, files, publisher_wallet)
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/ocean/ocean_assets.py:326: in _create1
    (data_nft, datatokens, asset) = self.create(
ocean_lib/ocean/ocean_assets.py:599: in create
    data_nft.set_metadata(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/models/data_nft.py:88: in set_metadata
    return self.send_transaction(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/web3_internal/contract_base.py:193: in send_transaction
    return contract_function.transact(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/web3_internal/web3_overrides/contract.py:74: in transact
    return transact_with_contract_function(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/web3_internal/web3_overrides/contract.py:143: in transact_with_contract_function
    wait_for_transaction_receipt_and_block_confirmations(
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/enforce_typing/decorator.py:29: in wrapper
    return func(*args, **kwargs)
ocean_lib/web3_internal/web3_overrides/utils.py:35: in wait_for_transaction_receipt_and_block_confirmations
    receipt = web3.eth.wait_for_transaction_receipt(tx_hash, transaction_timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <web3.eth.Eth object at 0x7f3c16fc50a0>
transaction_hash = HexBytes('0xf7e9c54c1c6e3b6830daddf8351a281989744d5c50e25e91ccec957d09ce7fc2')
timeout = 60, poll_latency = 0.1

    def wait_for_transaction_receipt(
        self, transaction_hash: _Hash32, timeout: float = 120, poll_latency: float = 0.1
    ) -> TxReceipt:
        try:
            with Timeout(timeout) as _timeout:
                while True:
                    try:
                        tx_receipt = self._get_transaction_receipt(transaction_hash)
                    except TransactionNotFound:
                        tx_receipt = None
                    if tx_receipt is not None:
                        break
                    _timeout.sleep(poll_latency)
            return tx_receipt
    
        except Timeout:
>           raise TimeExhausted(
                f"Transaction {HexBytes(transaction_hash) !r} is not in the chain "
                f"after {timeout} seconds"
            )
E           web3.exceptions.TimeExhausted: Transaction HexBytes('0xf7e9c54c1c6e3b6830daddf8351a281989744d5c50e25e91ccec957d09ce7fc2') is not in the chain after 60 seconds

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/web3/eth.py:774: TimeExhausted
----------------------------- Captured stdout call -----------------------------
alice_wallet.address = '0x8bE6c8126C4383745c25F3b3Ca76D73E3Caf8544'
bob_wallet.address = '0xD6764a9F3387B14692CC783BcF5604a23EBb779f'
Call create_url_asset(), and wait for it to complete...
----------------------------- Captured stderr call -----------------------------
2022-10-15 13:20:53 fv-az119-792 ocean[9322] INFO Successfully created NFT with address 0x3C0Fbc60fa9730cd5771245C11313f151CCd6436.
2022-10-15 13:21:08 fv-az119-792 ocean[9322] INFO Successfully created datatoken with address 0x3240f147311b8E98146fFD8A6D248F7d7EbCb7d7.
2022-10-15 13:21:19 fv-az119-792 ocean_lib.data_provider.data_encryptor[9322] INFO Asset urls encrypted successfully, encrypted urls str: 0x048a2f6..500ffd6a52, encryptedEndpoint https://v4.provider.polygon.oceanprotocol.com/api/services/encrypt
2022-10-15 13:21:19 fv-az119-792 ocean_lib.data_provider.data_encryptor[9322] INFO Asset urls encrypted successfully, encrypted urls str: 0x046a8e7..20e218, encryptedEndpoint https://v4.provider.polygon.oceanprotocol.com/api/services/encrypt
------------------------------ Captured log call -------------------------------
INFO     ocean:ocean_assets.py:461 Successfully created NFT with address 0x3C0Fbc60fa9730cd5771245C11313f151CCd6436.
INFO     ocean:ocean_assets.py:546 Successfully created datatoken with address 0x3240f147311b8E98146fFD8A6D248F7d7EbCb7d7.
INFO     ocean_lib.data_provider.data_encryptor:data_encryptor.py:58 Asset urls encrypted successfully, encrypted urls str: 0x048a...ffd6a52, encryptedEndpoint https://v4.provider.polygon.oceanprotocol.com/api/services/encrypt
INFO     ocean_lib.data_provider.data_encryptor:data_encryptor.py:58 Asset urls encrypted successfully, encrypted urls str: 0x046a.....0e218, encryptedEndpoint https://v4.provider.polygon.oceanprotocol.com/api/services/encrypt

=========================== short test summary info ============================
FAILED tests/integration/remote/test_polygon.py::test_ocean_tx__create_url_asset
= 1 failed, 233 passed, 7 skipped, 8 deselected, 1 warning in 1452.48s (0:24:12) =
Error: Process completed with exit code 1.
```

**Comment by AnaLoznianu** [comment]
@calina-c Can we close this issue? I see you referenced a merged PR. Is the problem fixed?

**Comment by calina-c** [comment]
Closed by #1075
